pygments/lexers/modula2.py

```diff
@@ -1563,9 +1563,20 @@ def get_tokens_unprocessed(self, text):
     def analyse_text(text):
         """Not much we can go by. (* for comments is our best guess."""
         result = 0
+
+        is_pascal_like = 0
         if '(*' in text and '*)' in text:
-            result += 0.01
+            is_pascal_like += 0.5
         if ':=' in text:
-            result += 0.01
+            is_pascal_like += 0.5
+
+        if is_pascal_like == 1:
```

Comment:
Looks good, but can you make this a boolean flag (`pascal = x in text and y in text and z in text`)?  The float-adding-and-comparing seems to be unneeded...

Reply:
Sure!